ci/test_wheel_distributed_ucxx.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+set -euo pipefail
+
+PROJECT_NAME="distributed_ucxx"
+
+source "$(dirname "$0")/test_utils.sh"
+source "$(dirname "$0")/test_common.sh"
+
+mkdir -p ./dist
+RAPIDS_PY_CUDA_SUFFIX="$(rapids-wheel-ctk-name-gen ${RAPIDS_CUDA_VERSION})"
+RAPIDS_PY_WHEEL_NAME="${PROJECT_NAME}_${RAPIDS_PY_CUDA_SUFFIX}" rapids-download-wheels-from-s3 ./dist
+
+# Install previously built ucxx wheel
+RAPIDS_PY_WHEEL_NAME="ucxx_${RAPIDS_PY_CUDA_SUFFIX}" rapids-download-wheels-from-s3 ./local-ucxx-dep
+python -m pip install ./local-ucxx-dep/ucxx*.whl
+
+# echo to expand wildcard before adding `[extra]` requires for pip
+python -m pip install $(echo ./dist/${PROJECT_NAME}*.whl)[test]
+
+# TODO: We need distributed installed in developer mode to provide test utils,
+# we still need to match to the `rapids-dask-dependency` version.
```

Comment:
@pentschev Can you explain a bit more about this requirement, and how you would resolve the TODO here? It looks like this is a requirement to run distributed-ucxx tests, but it isn't available in the `distributed` package? Having a dependency on installing `distributed` from source (as opposed to something available in the wheel) might make it hard to reproduce CI failures in a local development environment.

Reply:
Let's discuss here instead: https://github.com/rapidsai/ucxx/pull/167#discussion_r1470179504

Reply:
Just wrote https://github.com/rapidsai/ucxx/pull/167#discussion_r1470179504 about this. Yes, we need some testing dependencies from distributed, specifically https://github.com/rapidsai/ucxx/blob/94a19fde0e718ab3af18e1afc5ba79988dee9fdd/python/distributed-ucxx/distributed_ucxx/tests/test_ucxx.py#L173 . I don't know of a good solution for this problem either though and I very much don't want to copy-paste the code in here because that will eventually cause us to get out-of-sync with Distributed code and essentially mean we're not testing it properly anymore.